modules/solid_mechanics/src/materials/crystal_plasticity/ComputeElasticityTensorCP.C

```diff
@@ -56,6 +60,22 @@ ComputeElasticityTensorCP::ComputeElasticityTensorCP(const InputParameters & par
        (parameters.isParamSetByUser("euler_angle_3"))))
     mooseError("Bunge Euler angle information and the rotation_matrix cannot both be specified. "
                "Provide only one type of orientation information in the input file.");
+
+  // Check if source of Euler angle values has a conflict
+  if (_read_prop_user_object && _n_euler_angle_vars)
+    paramError("euler_angle_variables",
+               "Euler angles cannot be supplied from both coupled variables and auxiliary input "
+               "file in the option `read_prop_user_object`.");
+
+  if (isCoupled("euler_angle_variables") && _n_euler_angle_vars != 3)
+    paramError("euler_angle_variables", "The Euler angles should have three components.");
+
+  if (isCoupled("euler_angle_variables"))
+  {
+    for (unsigned int i = 0; i < _n_euler_angle_vars; i++)
+      _euler_angle_vars.push_back(
+          dynamic_cast<MooseVariable *>(getVar("euler_angle_variables", i)));
+  }
```

Comment:
we have `coupledValues` to do this with one call

Reply:
https://github.com/idaholab/moose/blob/8465925da35e576d75247f34c562ed4889214cc0/framework/include/interfaces/Coupleable.h#L205

Reply:
Thanks for the suggestions - yes the coupledValues is good for this case. Addressed